test/viewmodels/map/map_pin_view_model_integration_test.dart

```diff
@@ -19,22 +27,37 @@ void main() {
     late GeolocationService geolocationService;
     late ProviderContainer container;
     late FirestorePostGenerator postGenerator;
+    late MockUserRepositoryService userRepositoryService;
+    late ProviderScope mapWidgetWithPins;
 
     setUp(() async {
       fakeFireStore = FakeFirebaseFirestore();
       mockGeolocator = MockGeolocatorPlatform();
       geolocationService = GeolocationService(geoLocator: mockGeolocator);
       postGenerator = FirestorePostGenerator();
+      userRepositoryService = MockUserRepositoryService();
+      mapWidgetWithPins = newMapPageWithPinsNoOverride(
+        geolocationService,
+        fakeFireStore,
+        testingUserFirestoreId,
+        userRepositoryService,
+      );
 
       when(mockGeolocator.isLocationServiceEnabled())
           .thenAnswer((_) async => true);
       when(mockGeolocator.checkPermission())
           .thenAnswer((_) async => LocationPermission.always);
 
+      when(userRepositoryService.getUser(testingUserFirestoreId))
+          .thenAnswer((_) async => testingUserFirestore);
+
       container = ProviderContainer(
         overrides: [
-          firestoreProvider.overrideWithValue(fakeFireStore),
           geolocationServiceProvider.overrideWithValue(geolocationService),
+          loggedInUserIdProvider.overrideWithValue(testingUserFirestoreId),
+          firestoreProvider.overrideWithValue(fakeFireStore),
+          userRepositoryServiceProvider
+              .overrideWithValue(userRepositoryService),
```

Comment:
By humble curiosity, why is this necessary when you already have `loggedInUserIdProvide.overrideWithValue(testingUserFirestore)`?

Reply:
because the map pin viewmodel also needs the user info and not only the user ID

Reply:
Could `loggedInUserIdProvide.overrideWithValue(testingUserFirestore)` be removed then?

Reply:
No because it need this id to get the rigth user info in the user repo

Reply:
Makes sense, thank you ! 

In fact, after the test refactor, this was no longer even applicable.